src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -566,37 +566,64 @@ public bool NeedsNullableAttribute()
             return (object)type != null;
         }
 
-        public void AddNullableTransforms(ArrayBuilder<byte> transforms)
+        /// <summary>
+        /// If the type is a non-generic value type or Nullable&lt;&gt;, and
+        /// is not a type parameter, the nullability is not included in the byte[].
+        /// </summary>
+        private static bool SkipNullableTransform(TypeSymbol type)
         {
-            var typeSymbol = Type;
-            byte flag;
-
-            if (NullableAnnotation.IsOblivious() || typeSymbol.IsValueType)
+            var namedType = type as NamedTypeSymbol;
+            if (namedType is null)
             {
-                flag = NullableAnnotationExtensions.ObliviousAttributeValue;
+                return false;
             }
-            else if (NullableAnnotation.IsAnnotated())
+            if (namedType.IsGenericType)
             {
-                flag = NullableAnnotationExtensions.AnnotatedAttributeValue;
+                return type.IsNullableType();
             }
-            else
+            return type.IsValueType && !type.IsTupleType;
```

Comment:
>type.IsValueType [](start = 19, length = 16)

Are we testing scenarios when concrete value type references within another type cannot be resolved from metadata (an assembly reference is missing, for example)? It looks like this might cause a confusion during ```ApplyNullableTransforms``` causing us to apply flags to wrong places.